modules/targetanalysis/src/pixel_object_list.cpp

```diff
@@ -0,0 +1,271 @@
+            //Skip analysis to save computation time
+            continue;
+        }
+
+        for (int i = 0; i < numIntervals; i++){
+            //Reposition the contours for Mat operations (in the positive quadrant)
+            Point matReference(radius1-center1.x > radius2-center2.x?radius1-center1.x:radius2-center2.x, radius1-center1.y > radius2-center2.y?radius1-center1.y:radius2-center2.y);
+            for (int j = 0; j < v1Mod.size();j++){
+                v1Mod[j] = v1Mod[j] + matReference;
+            }
+            for (int j = 0; j < v2Mod[i].size(); j++){
+                v2Mod[i][j] = v2Mod[i][j] + matReference;
+            }
+            vector<vector<Point>> contoursWrapperA, contoursWrapperB;
+            contoursWrapperA.push_back(v1Mod);
+            contoursWrapperB.push_back(v2Mod[i]);
+                
+            //No point comparing contours if there is only one.
+            if (listLength >= 1){
+                double sim = compare_contours(contoursWrapperA, contoursWrapperB);
+                if (maxSimilarity < sim){
+                    BOOST_LOG_TRIVIAL(debug) << "Found new optimized position: " << sim;
+                    maxSimilarity = sim;
+                    minimumContour = v2Mod[i];
+                        
+                   /*//Visualization
+                    if (sim > 0.7){
+                        Mat drawing = Mat::zeros(matReference.x*2, matReference.y*2, CV_8UC3);
+
+                        drawContours(drawing, contoursWrapperA, 0, cv::Scalar(255,0,255), FILLED);
+                        drawContours(drawing, contoursWrapperB, 0, cv::Scalar(0,255,255), FILLED);
+ 
+                        namedWindow( "Duplicate Contours", CV_WINDOW_AUTOSIZE );
+                        imshow( "Duplicate Contours", drawing );
+                        waitKey(0);
+                    }*/   
+                }
+            }
+        }
+    } 
+    return maxSimilarity;
+}
+
+void PixelObjectList::addAndCompare(PixelObject* po){
+    //Iterate over list
+    int i = 0;
+    struct comparitor listMatch[listLength];
+    double maxSimilarity = 0;
+    int iMax = 0;
+
+    poNode* tempPointer;
+    tempPointer = tail;
+    BOOST_LOG_TRIVIAL(debug) << "List Length: " << listLength;
+    while(i<listLength){
+        //Returns similarity percentage - 1 is an ideal match, 0 is the worst
+        //match possible
+        listMatch[i].similarity = compareNode(po, tempPointer->o);
+        BOOST_LOG_TRIVIAL(debug) << "Compare Done";
+        listMatch[i].node = tempPointer;
+
+        //Update the best match if a better one is available
+        if (listMatch[i].similarity > maxSimilarity){
+            maxSimilarity = listMatch[i].similarity;
+            iMax = i;
+        }
+        i++;
+        tempPointer = tempPointer->next;
+    }
+    //Is the best match good enough - FUZZY LOGIC
+    //TODO: This should be a configurable value via a socket connection or a
+    //learning algorithm
```

Comment:
Little slow on this PR. This TODO would be something I'll try and include in a separate PR.